packages/core/src/queries/user.ts

```diff
@@ -98,15 +98,17 @@ export const countUsers = async (search?: string) =>
   envSet.pool.one<{ count: number }>(sql`
     select count(*)
     from ${table}
-    ${conditionalSql(search, (search) => sql`where ${buildUserSearchConditionSql(search)}`)}
+    where not (${fields.roleNames}::jsonb?${UserRole.Admin})
+    ${conditionalSql(search, (search) => sql`and ${buildUserSearchConditionSql(search)}`)}
```

Comment:
@logto-io/eng 
Should we filter the admin role at the query level?  As the query method name itself does not indicate it's a non-admin user query.

Reply:
Now provides a filter option `hideAdminUser`, PTAL